src/catalua.h

```diff
@@ -3,9 +3,67 @@
 #define CATALUA_H
 
 #include "int_id.h"
+#include "enums.h"
+#include "item.h"
 
 #include <string>
 #include <sstream>
+#include <vector>
+#include <list>
+
+#ifdef LUA
+extern "C" {
+#include "lua.h"
+#include "lualib.h"
+#include "lauxlib.h"
+}
+#endif //LUA
+
+enum CallbackArgumentType : int {
+    Undefined = -1,
+    Integer = 0,
+    Number,
+    Double = Number,
+    Float = Number,
+    String,
+    Tripoint,
+    Item
+};
+
+struct CallbackArgument {
+    std::string name = "";
+    CallbackArgumentType type = CallbackArgumentType::Undefined;
+
+    int value_integer = 0;
+    float value_number = 0.0f;
+    std::string value_string = "";
+    tripoint value_tripoint = tripoint( 0, 0, 0 );
+    item value_item = item();
+
+    CallbackArgument( const std::string &arg_name, int arg_value ) : name( arg_name ),
+        type( CallbackArgumentType::Integer ), value_integer( arg_value ) {
+    }
+    CallbackArgument( const std::string &arg_name, double arg_value ) : name( arg_name ),
+        type( CallbackArgumentType::Number ), value_number( ( float ) arg_value ) {
```

Comment:
Please don't use C-style casts, they can be dangerous. Always use a (more specific and limited) C++ cast. In this case a `static_cast<float>` would do. But why cast this manually anyway, why not let the compiler convert it implicitly?